File: atv11/exerc11.go
package main
import "fmt"

type Funcionario interface {
    CalcularSalario() float64
}

type FuncionarioHorista struct {
    Nome            string
    HorasTrabalhadas float64
    ValorHora       float64
}

func (f FuncionarioHorista) CalcularSalario() float64 {
    return f.HorasTrabalhadas * f.ValorHora
}

type FuncionarioAssalariado struct {
    Nome       string
    SalarioFixo float64
}

func (f FuncionarioAssalariado) CalcularSalario() float64 {
    return f.SalarioFixo
}

func main() {
    f1 := FuncionarioHorista{"João", 160, 20}
    f2 := FuncionarioAssalariado{"Maria", 3000}

    fmt.Println(f1.Nome, "recebe: R$", f1.CalcularSalario())
    fmt.Println(f2.Nome, "recebe: R$", f2.CalcularSalario())
}

File: atv11/exerc11.py
from abc import ABC, abstractmethod

class Funcionario(ABC):
    def __init__(self, nome):
        self.nome = nome

    @abstractmethod
    def calcular_salario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, nome, horas_trabalhadas, valor_hora):
        super().__init__(nome)
        self.horas_trabalhadas = horas_trabalhadas
        self.valor_hora = valor_hora

    def calcular_salario(self):
        return self.horas_trabalhadas * self.valor_hora

class FuncionarioAssalariado(Funcionario):
    def __init__(self, nome, salario_fixo):
        super().__init__(nome)
        self.salario_fixo = salario_fixo

    def calcular_salario(self):
        return self.salario_fixo

f1 = FuncionarioHorista("João", 160, 20)
f2 = FuncionarioAssalariado("Maria", 3000)

print(f"{f1.nome} recebe: R${f1.calcular_salario()}")
print(f"{f2.nome} recebe: R${f2.calcular_salario()}")
